### PR TITLE
General Improvements

### PR DESCRIPTION
**Pull request by pjlast**
This PR adds some general improvements/cleanups:

1. Don't automatically go into insert mode when opening CodyChat
2. Detect repo name and set codebase during agent initialize
3. Process context files in chat responses and display them to the user
  3.1. Context files are folded by default
4. Update to latest version of agent

I also tried to add some better type annotations to some of our messages/callback functions

image

### CODE REVIEW

File: lua/sg/cody/rpc.lua [hunk 1]
```diff
@@ -170,6 +175,25 @@ M.request = async.wrap(function(method, params, callback)
   end)
 end, 3)
 
+---@return string | nil
+local function get_current_repo_url()
```
Comment (tjdevries) [inline lua/sg/cody/rpc.lua:179]: I have something like this in lua/sg/cody/context.lua, so we should probably just consolidate them somewhere

File: lua/sg/cody/state.lua [hunk 2]
```diff
@@ -61,8 +61,9 @@ end
 --- Get a new completion, based on the state
 ---@param bufnr number
 ---@param win number
----@param callback function(noti)
-function State:complete(bufnr, win, callback)
+---@param code_only boolean
+---@param callback CodyChatCallback
+function State:complete(bufnr, win, code_only, callback)
```
Comment (tjdevries) [inline lua/sg/cody/state.lua:66]: i think `code_only` should be `opts` and we can have a field for `code_only`. Cause we'll probably add more stuff for this later.

File: lua/sg/components/layout/hover.lua [hunk 3]
```diff
@@ -102,14 +102,20 @@ end
...
-    local lines = vim.split(noti.text, "\n")
-    if code_only then
-      table.remove(lines, 1)
-      table.remove(lines)
+  self.state:complete(self.history.bufnr, self.history.win, code_only, function(msg)
+    local lines = vim.split(msg.text, "\n")
+    local render_lines = {}
+    for _, line in ipairs(lines) do
+      if code_only then
+        if vim.trim(line) == "```" then
```
Comment (tjdevries) [inline lua/sg/components/layout/hover.lua:110]: should this match against the line? Does it ever send like "```lua" ?

Comment (pjlast) [reply lua/sg/components/layout/hover.lua:110]: Cody's prompt gets prefilled with "```lua" so it's omitted from the start of the LLM's response. And the assumption I'm making here is that a line that's only "```" will be the end of the code response.

But this isn't perfect. I think using some xml tag shenanigans for code blocks would work better, but would require updating agent again.

**Pull request merged**